barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -4387,6 +4388,8 @@ std::vector<Row> AvmTraceBuilder::finalize(uint32_t min_trace_size, bool range_c
         // check the clock and iterate through the main trace until we hit the clock
         auto clk = src.clk;
 
+        main_trace.at(1).avm_kernel_side_effect_counter = initial_side_effect_counter;
```

Comment:
The `at(1)` is a bit mysterious, could you add a comment explaining?

Why is this inside the loop?

Reply:
at(1) is because in the current code the first assignment happens for j = 1 below. So, setting the first prev.avm_kernel_side_effect_counter essentially.

Inside the loop: because I am an idiot :-) and tired. Will fix tomorrow.